### PR TITLE
Release engagement cohorts

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

This PR prepares the release of `engagement-cohorts-4349` for everyone by removing feature flagged code and legacy code. While the feature has been available for several weeks now, it was only released to beta users. Today (Sep 8) I released to 50% of cloud users to get some feedback and monitor usage for a couple of days before merging this in.

PR will also enable anyone on Postgres to get the new UI for cohorts. Only excluded functionality is creating an action/event based cohort with a number of occurrences selector.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: frontend/src/scenes/trends/personsModalLogic.ts [hunk 1]
```diff
@@ -212,17 +211,6 @@ export const personsModalLogic = kea<personsModalLogicType<PersonModalParams>>({
         },
     }),
     listeners: ({ actions, values }) => ({
-        refreshCohort: () => {
```
Comment (paolodamico) [inline frontend/src/scenes/trends/personsModalLogic.ts:215]: Looks like this wasn't doing anything useful as we remain on the person modal after saving a cohort, but would appreciate some confirmation.

File: frontend/src/scenes/cohorts/Cohort.tsx [hunk 2]
```diff
@@ -1,203 +1,215 @@
 import React from 'react'
-import { CohortGroup } from './CohortGroup'
-import { cohortLogic } from './cohortLogic'
-import { Button, Card, Col, Divider, Input, Row } from 'antd'
-import { AimOutlined, ArrowLeftOutlined, InboxOutlined, UnorderedListOutlined } from '@ant-design/icons'
-import { useValues, useActions, BuiltLogic } from 'kea'
+import { useActions, useValues } from 'kea'
```
Comment (paolodamico) [inline frontend/src/scenes/cohorts/Cohort.tsx:2]: Change is we actually moved `CohortV2.tsx` to here and dropped the legacy `Cohort.tsx` altogether.

**Comment by paolodamico** [comment]
We should definitely address #5852 before merging this

**Comment by timgl** [comment]
Thanks for doing this. I spent a little bit of time doing the same yesterday and realised CSV uploads were broken in the new version (it basically expects groups to be set even with CSV). just FYI, i can have a look at fixing this later as well

**Comment by paolodamico** [comment]
Np and thanks @timgl! I can also get back to this tomorrow if needed

**Comment by timgl** [comment]
@paolodamico have fixed most issues, mind taking a look?

**Comment by paolodamico** [comment]
Fixed one little bug with the tooltip but looks good to go @timgl! Merging when tests pass

**Pull request merged**